CollAction/Controllers/ProjectsController.cs

```diff
@@ -186,12 +197,23 @@ public ViewResult StartInfo()
             foreach (var admin in administrators)
                 await _emailSender.SendEmailAsync(admin.Email, subject, confirmationEmailAdmin);
 
-            return View("ThankYouCreate", new ThankYouCreateProjectViewModel()
-            {
-                Name = project.Name
-            });
+            string validUrlForProjectName = WebUtility.UrlEncode(project.Name);
+            if(String.IsNullOrEmpty(validUrlForProjectName))
+             {
+                return NotFound();
+            }
+            return LocalRedirect("~/Projects/Create/"+validUrlForProjectName+"/thankyou");
         }
 
+        [Authorize]
+        public IActionResult ThankYouCreate(string name)
+        {
+            return View(new ThankYouCreateProjectViewModel
+            {
+                Name = WebUtility.UrlDecode(name)
```

Comment:
The project name should already be decoded.

Reply:
Apparently due to some reason this is not always the case ( checked it ), so it's better to decode it always. Could have something to do with redirect.